rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -133,7 +133,12 @@ def _query_web3_get_logs(
                 msg = 'query returned more than 10000 results'
 
             # errors from: https://infura.io/docs/ethereum/json-rpc/eth-getLogs
-            if msg in {'query returned more than 10000 results', 'query timeout exceeded'}:
+            if msg in {
+                'query returned more than 10000 results',
+                'eth_getLogs is limited to a 1000 blocks range',  # seen in https://1rpc.io/gnosis
+            }:
+                block_range = 999
```

Comment:
Why force 999? If the node can handle less than 10k then let it return 9999. If the node can handle less than 1k then do 999. You do the conditional check anyway. Use it.